### ISSUE TITLE
Tesseract security tab configuration not expanding

### ISSUE THREAD

**Issue by libix** [CoFH/Feedback#281]
Hello !

I've looked in the other posts related to tesseract, but my problem is quite different.
When i lock my tesseract (with signalum lock in a crafting table) an I place it to the ground, everything works, but in the GUI, the last tab (the security one) don't expand when i click on it (left/right/shift click).
I tried desactivating NEI, but it doesn't change anything, and all the others tabs are working (redstone configuration, send/receive, tutorial...)

I have this bug since update Thermal Expansion 4.0.0 RC6-138, and I even tried the alpha release from curseforge (RC6-139)
My forge and all the cofh mods are up to date.

Am I missing something, or somebody else has this issue too ?

**Comment by KingLemming** [comment]
Have you changed your name recently?

**Comment by libix** [comment]
no i didnt, and the tesseracts i crafted before updating are still under my control (owner only) i can edit them too. But in all my tesseracts (old and new) , the the security tab dont work when i click.

**Comment by KingLemming** [comment]
I can't reproduce this at all.

**Comment by libix** [comment]
no it's stil in the same world, i will check again if there are some updates i missed,
it's maybe due to a conflict with other mods ( i updated botania, simply Jetpacks and all the thermal mods at the same time).

**Comment by libix** [comment]
ok, so I made a few tests
in a solo world, all the tabs are working.
On my server, I'm not allowed to modify the security tab only, I still can configure the tesseract itself.
If the server dont allow me to modify the security, I shoud not be allowed to change the inputs/outputs or the redstone behavior, that's weird.
Is their anyway to "reset" the links between players from all the server and their privated tesseracts ? (a file to delete or something  ?)

**Comment by KingLemming** [comment]
I still can't make this happen. And there is a file in the world save folder, yes. Just poke around and it should be pretty obvious which.

**Comment by libix** [comment]
It's a problem related to my server.
When changing my username, i can't right click the tesseracts (they are secured by signalum, so it's normal) but the new ones I craft still have the security tab bug.
Since that bug isn't showing in my single player world, and occured on my map only after the update to RC6 (coming from RC2), it might have corrupt something in my server.

**Comment by KingLemming** [comment]
Changing your username? We update correctly for that unless you're circumventing it locally with some other mod.

**Comment by BreeOge** [comment]
Libix your not alone, I too am having the same exact problem...  it started with a recent update.. i think around 2 updates ago..

**Comment by libix** [comment]
yeah it was something like 2 updates ago, but does your problem happen in solo world ?

**Comment by BreeOge** [comment]
Dont play solo, so I havent tried.. i will try it and get back to you..

**Comment by xxTheDoctor** [comment]
I also am having this problem. I'm not quite sure whether it was the result of an update to the mod or to me changing my Minecraft profile name.

As of February (I think), Minecraft introduced profile name changes. I've noticed that when I create a locked tesseract and place it, the tesseract says the owner is my old, and original, profile name. All of my old tesseracts that used to work, I can no longer access the privacy tab to change the settings. Those tesseracts, as well as the new ones, all display the owner as my original name.

I've tested the tesseracts in Single Player and on a server and the problem persisted. I tested it on a FTB Infinity 1.20 server and Single Player and 1.21 Single Player.

**Comment by xxTheDoctor** [comment]
Crash Logs

[23:33:57] [Client thread/FATAL]: Unreported exception thrown!
java.lang.NullPointerException
	at net.minecraft.server.management.PreYggdrasilConverter.func_152719_a(SourceFile:244) ~[nz.class:?]
	at cofh.thermalexpansion.block.TileInventory.setOwnerName(TileInventory.java:346) ~[TileInventory.class:?]
	at cofh.thermalexpansion.block.ender.BlockEnder.func_149689_a(BlockEnder.java:76) ~[BlockEnder.class:?]
	at net.minecraft.item.ItemBlock.placeBlockAt(ItemBlock.java:213) ~[abh.class:?]
	at net.minecraft.item.ItemBlock.func_77648_a(ItemBlock.java:103) ~[abh.class:?]
	at net.minecraft.item.ItemStack.func_77943_a(ItemStack.java:129) ~[add.class:?]
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_78760_a(PlayerControllerMP.java:360) ~[bje.class:?]
	at net.minecraft.client.Minecraft.func_147121_ag(Minecraft.java:1435) ~[bao.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1941) ~[bao.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:961) ~[bao.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:887) [bao.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:148) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.7.0_72]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.7.0_72]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.7.0_72]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.7.0_72]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.11.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.11.jar:?]

**Comment by KingLemming** [comment]
Infinity is way out of date. That's the problem there.

**Comment by Etheras** [comment]
Just noticed i have this issue today and updated from RC6 to RC7, but was still present.
Personal collection of mods, no modpack, all updated to latest.
Noticed server has "CoFHCore.cfg" and "core/common.cfg" so maybe it has something to do with switching cfg's around at some update ?
Because the new "core/friends.cfg" is empty but the old "CoFHSocial-Friends.cfg" was populated with the list.

**Comment by Lightirius** [comment]
Same thing. Cannot open security tab after update from RC2 to RC7 on existing world. To reproduse:
1)Install COFHCore RC2, ThermalFoundation RC3, ThermalExpansion RC2.
2)Create world
3)Craft and place tesseract with security upgrade (don't test on other machines)
4)Update all mods to RC7

Tested on server on clean install with only ThermalExpansion, ThermalFoundation, COFHCore, NEI, CodeChickenCore.
ArchLinux, Oracle Java 8-u40

**Comment by Lightirius** [comment]
More tests:
Security tab not work after CofhCore RC2, TE RC2, ThermalFoundation RC3 on standalone server, but working fine in single player and open to lan (clean install with only this mods and NEI).
Tested on ArchLinux/Windows 10, Java 7/8, forge 1291/1343

**Comment by Lightirius** [comment]
OK, found solution. Players UUID's with online-mode=false differ from UUID's with online-mode=true for some reason. Start server with online-mode=true and craft new machine with security upgrade fix this issue

**Comment by Etheras** [comment]
What Lightirius said "works" in the term that, "offline UUID" tesseracts are still bugged(new or old), but tesseracts created with the "online UUID" are accessible even from the "offline UUID".

**Comment by skyboy** [comment]
Well, that gives me enough information to fix the issue. Though the issue is hilarious.

**Comment by KingLemming** [comment]
Okay so the question is...why were you guys ever playing on a server that wasn't authenticating people?

**Comment by Lightirius** [comment]
I really don't need this thing. I play with my friends throw hamachi ( or local lan some times) and server is "hosted" on my computer.

**Comment by tetrisdaemon** [comment]
There are quite a few use cases for offline mode, outside of cracked servers. For example, developers may need to multi-client in order to test their mods, running a server in offline mode so they wouldn't need to buy multiple copies of the game. 

And, as I'm sure you're aware of, being the seasoned developer you are, authentication is completely doable on an offline server. 

The problem here then isn't that the clients are to blame for playing on offline mode, or even a server promoting itself as cracked. The problem is that the product has a fault, and those in charge of it are responsible for fixing it.

**Comment by skyboy** [comment]
We have no requirement of offline mode or buying multiple copies of the game when we have access to the code. There's a reason most in-dev screenshots feature the steve skin. That argument is entirely nonsense.

The problem here is specifically that clients (unless hacked) are in online mode, while the server is in offline mode causing the data we compare to be desynched, and as stated, it will be fixed.

Additional people making aggressive posts about this are going to delay me fixing it.

**Comment by tetrisdaemon** [comment]
I wasn't aware of that, since I had assumed that developers were running in offline mode during testing. Nevertheless, there's the certain use case of cracked servers, which carry a non-negligible weight in your player base.

Allow me to clarify the initial intention of my post. It wasn't meant to be aggressive, and I didn't mean for it to offend; I had directed it at KingLemming for questioning why players would play on an offline-mode server.

In any case, why would people making aggressive posts delay your fixing it? Many have praised you for your success (and your mods are quite popular), so expect the contrary when you fail.

**Comment by skyboy** [comment]
This is a particularly niche issue, and very few people are affected by it, making it lower priority. A loud (and aggressive) minority will only make me prefer to work on other things, more pressing or otherwise.

**Comment by KingLemming** [comment]
>The problem here then isn't that the clients are playing on offline mode, or even on a server promoting itself as cracked. The problem is that the product has a fault, and those in charge of it are responsible for fixing it.

It's not quite that simple. For the same reason that we don't have capes anymore, we have to be *really* careful about what Mojang is kosher with. And open support of potential pirate servers is sketchy.

Also, it's not a product really. We do this for fun and we have never promised anything. Would I like it fixed? Sure. Would I like the peace of mind knowing that Mojang isn't going to destroy us for assisting pirates? Yes I'd like that too.

It'll happen when it's possible, but it really is low priority since the vast majority of servers require legitimate Mojang authentication.

**Comment by BreeOge** [comment]
I personally would like to see this fixed sooner than later honestly.. but thats just me.. As for the whole Online Offline issue, people make too big of a deal about it.. Simply put, if mojang didn't want an offline mode they could remove it.. but they haven't.. End of story..   But KingLemming, and Skyboy you guys make an awesome mod and I praise you guys for it..

**Comment by KingLemming** [comment]
So, this *should* be fixed now.

**Comment by Etheras** [comment]
Thank you.

**Comment by BreeOge** [comment]
Thank you
